### PR TITLE
Add getState and setState methods in RestateTestEnvironment

### PR DESCRIPTION
**Pull request by jackkleeman**

### CODE REVIEW

File: templates/typescript-testing/test/restate_test_environment.ts [hunk 1]
```diff
@@ -109,6 +110,68 @@ export class RestateTestEnvironment {
...
+          }
+      );
+
+      if (!res.ok) {
+          const badResponse = await res.text();
+          throw new Error(
+              `Error ${res.status} during modify state: ${badResponse}`
+          );
+      }
+    }
```
Comment (slinkydeveloper) [inline templates/typescript-testing/test/restate_test_environment.ts:142]: for reference, In Java I just generate a client from our OpenAPI definition and publish it as separate module, and then use it in our `sdk-testing` module (the one with the testcontainers thing)

File: templates/typescript-testing/test/restate_test_environment.ts [hunk 2]
```diff
@@ -6,6 +6,7 @@ import {
     TestContainers,
     Wait,
 } from "testcontainers";
+import { tableFromIPC } from "apache-arrow";
```
Comment (slinkydeveloper) [inline templates/typescript-testing/test/restate_test_environment.ts:9]: Bringing in apache arrow here is such a PITA :( 

I think down the road we should have an option in our `/query` endpoint to simply return json, as I do in the CLI...

Comment (jackkleeman) [reply templates/typescript-testing/test/restate_test_environment.ts:9]: yes, i honestly wonder if i should just do this now

File: templates/typescript-testing/test/restate_test_environment.ts (hunk 1 above)
Comment (slinkydeveloper) [reply templates/typescript-testing/test/restate_test_environment.ts:142]: What I would hope is that in typescript SDK we eventually have an autogenerated package for the admin api. But yes you can ignore this comment now, nothing we should be doing for it.

**Pull request merged**